### DISCUSSION TITLE
Sandbox React App & React UI communication components in iframe

### DISCUSSION THREAD

**Discussion by codylindley** [Azure/communication-ui-library#4650] [Q&A]
We'd like to use the React communication components in a sandbox environment (i.e., iframed into another application and controlled via post message using the correct CORS headings). 

1. Are you aware of anyone else doing this?
2. Is there anything about this architecture that you believe is a nonstarter.

**Comment by dmceachernmsft** [comment]
Hi @codylindley! Thanks for starting the discussion! To answer you questions:
1. We aren't currently aware of anyone doing this in production. If you are curious about seeing a calling application that sends the call to a new window using post messaging check out our Calling Widget sample [here](https://github.com/Azure-Samples/communication-services-web-calling-widget-hero). Its configured to be used in the same window out of the box but the functionality is there to light up the new window behaviour.
2. The only thing that comes to mind is you have to be careful with your Calling Client (Or CallAdapter if using our composites) you can't send a reference to the Calling client or adapter through the post messaging of the original window. Looking into the dataChannel there was an issue with passing references of the handlers and event subscriptions for the client as well. Current advise would be, you would want to create the client in the new window or iframe to prevent multiple clients made for the one user.
